crates/ruff_linter/src/rules/flake8_pyi/rules/duplicate_literal_member.rs

```diff
@@ -0,0 +1,68 @@
+use std::collections::HashSet;
+
+use rustc_hash::FxHashSet;
+
+use ruff_diagnostics::{Diagnostic, FixAvailability, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::comparable::ComparableExpr;
+use ruff_python_ast::Expr;
+use ruff_python_semantic::analyze::typing::traverse_literal;
+use ruff_text_size::Ranged;
+
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+/// Checks for duplicate literal members.
+///
+/// ## Why is this bad?
+/// Duplicate literal members are redundant and should be removed.
+///
+/// ## Example
+/// ```python
+/// foo: Literal[True, False, True]
+/// ```
+///
+/// Use instead:
+/// ```python
+/// foo: Literal[True, False]
+/// ```
```

Comment:
Let's use an example that doesn't have booleans in the `Literal` slice, since strictly speaking `Literal[True, False]` could be written more simply using `bool` anyway:

```suggestion
/// ```python
/// foo: Literal["a", "b", "a"]
/// ```
///
/// Use instead:
/// ```python
/// foo: Literal["a", "b"]
/// ```
```